### PR TITLE
Simplify integration test

### PR DESCRIPTION
**Pull request by gen740**
## Motivation

The first step of #30.

To divide the integration test for each integration, the GitHub workflow file must be kept simple and concise.

## Description of the changes

- Remove `mac-tests.yml`
- Remove `coverage.yml`
- Simplify `tests.yml`
  - Remove cache action.
  - Remove some cocurrency.
- Refactor tests to remove unused pytest integration label.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -71,21 +47,7 @@ jobs:
...
-
     - name: Tests
-      run: |
-        pytest tests \
-          --ignore tests/test_mxnet.py
-
-    - name: Tests(Deprecated)
       run: |
         pip install "numpy<1.24"
-        pytest tests/test_mxnet.py
```
Comment (y0z) [inline .github/workflows/tests.yml:91]: Is `test_mxnet.py` no longer used?
Then, can it be removed from the repository?

Comment (gen740) [reply .github/workflows/tests.yml:91]: This PR represents the initial phase of dividing integration tests. The tests will be separated into individual integrations. For simplicity, I have removed the process specific to a particular integration.
I believe this process should be implemented in the MXNet tests (e.g., `test_mxnet.yml`)

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -9,10 +9,6 @@ on:
     - cron: '0 23 * * SUN-THU'
   workflow_dispatch:
 
-concurrency:
```
[the remark this answers is not in the document]
Comment (gen740) [reply .github/workflows/tests.yml:12]: I wrote this based on the [Optuna example](https://github.com/optuna/optuna-examples/blob/main/.github/workflows/aim.yml).
I believe the CI should be as simple as possible.
What do you think?

[the remark this answers is not in the document]
Comment (y0z) [reply .github/workflows/tests.yml:12]: I see. No problem.

**Review by y0z — APPROVED** [review]
LGTM.

**Comment by eukaryo** [comment]
LGTM!

**Review by eukaryo — APPROVED** [review]

**Pull request merged**